lib/configDefaults.js

```diff
@@ -1,7 +1,30 @@
 const os = require('os')
 const path = require('path')
+const fs = require('fs')
 
-module.exports = {
+const reporter = require('./reporter')
+
+const userConfig = () => {
+    const configDir = path.join(os.homedir(), '.config', 'd2')
+    const configPattern = /^(d2\.)*(config)\.(js$|json$)/
+
+    try {
+        const files = fs.readdirSync(configDir)
+
+        // grab one of the config files if there are more than one
+        const configFile = files.filter(f => configPattern.test(f))[0]
```

Comment:
Could do a `Math.random()` check here to just load a file at random if there are multiple, but I think grabbing the first one in the array that matches is fine for now until we decide on the actual order of priority.

Reply:
I think I prefer explicit ordered list of "supported" config files, even if we use a simple array and `fs.exists` :

```js
const tryLoadConfig = file => {
  if (!fs.existsSync(file)) return null
  try {
    const userConfig = require(path.join(configDir, configFile))
    reporter.debug(
      `User configuration loaded from: ${configFile}`,
       userConfig
    )
    return userConfig
  } catch (e) {
    reporter.debug(`Failed to load user config from: ${configDir}`, e)
    return null
  }
}
const userConfig = () => {
  const files = [path.join(configDir, 'd2.config.js'), path.join(configDir, 'd2.config.json')]
  for (let f in files) {
    const config = tryLoadConfig(f)
    if (config) return config
  }
  return {}
}
```

This seems a bit clearer than randomly picking one if two exist, and has the added benefit of not restricting us to files in a single directory (though we're only using one now).  However, it has twice as many filesystem syscalls so may be a touch slower.

If you want to stick with the regex test that's fine, your call.

Reply:
(we could technically parallelize the fs.exist calls as async promises but that reeks of premature optimization)

Reply:
As written I think this logic will also happily load `~/.config/d2/undefined` because of the `[0]` de-reference on a potentially zero-length array.  Not a huge issue but could be a fun bug to track down later!

Reply:
Alright, I've adjusted it to your suggestion. It's simpler to understand. :+1:

Reply:
One mod: I dropped the `fs.exists()` call because it is deprecated: https://nodejs.org/docs/latest-v10.x/api/fs.html#fs_fs_exists_path_callback